packages/orm/src/changes.ts

```diff
@@ -31,7 +38,9 @@ export interface ManyToOneFieldStatus<T extends Entity> extends FieldStatus<IdOf
  *    i.e. `Publisher.changes` is typed as `Changes<Publisher, keyof Publisher | keyof SmallPub | keyof LargePub>`
  * @type R An optional list of restrictions, i.e for `Reacted` for to provide `changes` to its subset of fields.
  */
-export type Changes<T extends Entity, K = keyof FieldsOf<T>, R = K> = { fields: K[] } & {
+export type Changes<T extends Entity, K = keyof (FieldsOf<T> & RelationsOf<T>), R = K> = {
+  fields: NonNullable<K>[];
```

Comment:
Is the `NonNullable` required here? When would `K` be null/undefined?

Reply:
its because the `RelationsOf<T>` relies on the `OptsOf<T>` which can be `undefined`, so adding the non-nullable as a safety way to tell other casting types this fields won't include undefined

Reply:
Weird, I really can't explain why that's necessary, but :shrug: I guess we can ignore it for now.

Reply:
typescript gymnastics? 😕 i really don't know why it cast undefined as well with the other keys